src/plugins/grass/qgsgrassnewmapset.cpp

```diff
@@ -399,64 +399,29 @@ void QgsGrassNewMapset::setGrassProjection()
   }
 
   // Define projection
-  if ( !proj4.isEmpty() )
+  if ( crs.isValid() )
   {
-    QgsDebugMsgLevel( QString( "proj4 = %1" ).arg( proj4.toLocal8Bit().constData() ), 3 );
+    const QString wkt = crs.toWkt( Qgis::CrsWktVariant::Preferred );
+    QgsDebugMsgLevel( QStringLiteral( "wkt = %1" ).arg( crs.toWkt( Qgis::CrsWktVariant::Preferred ) ), 3 );
 
-    OGRSpatialReferenceH hCRS = nullptr;
-    hCRS = OSRNewSpatialReference( nullptr );
-    int errcode;
+    // Note: GPJ_osr_to_grass() defaults in PROJECTION_XY if projection
+    //       cannot be set
 
+    G_TRY
     {
-      QgsLocaleNumC l;
-      errcode = OSRImportFromProj4( hCRS, proj4.toUtf8() );
-    }
-
-    if ( errcode != OGRERR_NONE )
-    {
-      QgsDebugMsgLevel( QString( "OGR can't parse PROJ-style parameter string:\n%1\nOGR Error code was %2" ).arg( proj4 ).arg( errcode ), 2 );
-
-      mCellHead.proj = PROJECTION_XY;
-      mCellHead.zone = 0;
-      mProjInfo = nullptr;
-      mProjUnits = nullptr;
+      // There was a bug in GRASS, it is present in 6.0.x line
+      int ret = GPJ_wkt_to_grass( &mCellHead, &mProjInfo, &mProjUnits, wkt.toUtf8().constData(), 0 );
+      // Note: It seems that GPJ_osr_to_grass()returns always 1,
```

Comment:
```suggestion
      // Note: It seems that GPJ_osr_to_grass() returns always 1,
```

Still true?

Reply:
@lbartoletti that's all existing comments and code -- I don't feel confident removing those notes.